### PR TITLE
feat: add support for sslmode=disable

### PR DESCRIPTION
**Pull request by lucacasonato**
Adds support for sslmode "disable".

### CODE REVIEW

File: tests/connection_params_test.ts [hunk 1]
```diff
@@ -135,10 +136,10 @@ Deno.test("dsnStyleParametersWithInvalidSSLMode", function () {
   assertThrows(
     () =>
       createParams(
-        "postgres://some_user@some_host:10101/deno_postgres?sslmode=disable",
+        "postgres://some_user@some_host:10101/deno_postgres?sslmode=verify-full",
       ),
     undefined,
-    "Supplied DSN has invalid sslmode 'disable'. Only 'require' or 'prefer' are supported",
+    "Supplied DSN has invalid sslmode 'verify-full'. Only 'require' or 'prefer' are supported",
```
Comment (Soremwar) [inline tests/connection_params_test.ts:142]: Only 'require', 'prefer' and 'disable' are supported

File: connection/connection.ts [hunk 2]
```diff
@@ -255,56 +255,60 @@ export class Connection {
...
     await this.#createNonTlsConnection({ hostname, port });
 
-    const accepts_tls = await this.#serverAcceptsTLS()
-      .catch((e) => {
-        // Make sure to close the connection if the TLS validation throws
-        this.#conn.close();
-        throw e;
-      });
+    // If TLS is disabled, we don't even try to connect.
+    if (isTLSEnabled) {
```
Comment (Soremwar) [inline connection/connection.ts:267]: Per spec, all PostgreSQL servers with TLS encryption support (all supported versions) must start a connection asking whether of not the server is willing to create a TLS connection, even if the user requested TLS to be disabled. Simply move the "is TLS enabled" condition after the serverAcceptsTLS check to resolve this

**Review by Soremwar — CHANGES_REQUESTED** [review]

**Comment by twhitbeck** [comment]
we need this to be able to use this drive to connect to postgres instances on [fly](https://fly.io/docs/reference/postgres/)

**Review by Soremwar — APPROVED** [review]
LGTM, thank you Luca

**Pull request merged**